config/configclientauth/README.md

```diff
@@ -0,0 +1,54 @@
+# Client Authorization Settings
+
+This module provides HTTP/GRPC client types to be configured to perform authorization for requests using the following 
+
+## OAuth2 Client Credentials. (HTTP/gRPC)
+
+OAuth2 Client Credentials library exposes a [variety of settings](https://pkg.go.dev/golang.org/x/oauth2/clientcredentials)
+and implements the OAuth2.0 ["client credentials"](https://tools.ietf.org/html/rfc6749#section-1.3.4)
+token flow, also known as the ["two-legged OAuth 2.0" or machine to machine authorization](https://tools.ietf.org/html/rfc6749#section-4.4).
+
+This should be used when the client is acting on its own behalf.
+
+
+When enabled as a part of HTTP/gRPC Client Configuration, the corresponding Client with OAuth2 supported authorization transport is returned.
+The client manages the token auto refresh
+
+### OAuth2 Client Credentials Configuration
+
+- [`client_id`](https://tools.ietf.org/html/rfc6749#section-2.2): The unique identifier issued by authorization server to the registered client.
+- `client_secret`: Registered clients secret.
```

Comment:
```suggestion
- `client_secret`: Registered client's secret.
```

Reply:
You might want to add some warnings like the ones from https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/2341.